### PR TITLE
filter (ticdc): simplify filter interface and apply event filter rule in ddlPuller

### LINKED ISSUE #9531: rename tables trigger error and add Ignore-sql configured not effective, unable to skip the rename ddl
### What did you do?

1、changefeed configuration:
[filter]
**rules = ['cdctest.ta', 'cdctest.tb', 'cdctest.tc', 'cdctest.td']**

2、 rename tables on upstream:

rename table cdctest.tc to cdctest.tc_bak,cdctest.td to cdctest.tc,cdctest.tc_bak to cdctest.td;

3、cdc error:

[
  {
    "id": "test-ticdc-tatb-task",
    "namespace": "default",
    "summary": {
      "state": "error",
      "tso": 443433743909650434,
      "checkpoint": "2023-08-09 15:18:10.378",
      "error": {
        "time": "2023-08-09T15:29:16.383357628+08:00",
        "addr": "[172.16.201.141:8335](http://172.16.201.141:8335/)",
        "code": "CDC:ErrSnapshotTableNotFound",
        "message": "[CDC:ErrSnapshotTableNotFound]table 111 not found in schema snapshot"
      }
    }
  }
]

4、add ignore-sql configuration :
ignore-sql = ["^rename"]
or
ignore-sql = ["rename table cdctest.tc to cdctest.tc_bak,cdctest.td to cdctest.tc,cdctest.tc_bak to cdctest.td"]

5、update changefeed ：

/home/tidb/.tiup/components/ctl/v6.5.3/cdc cli changefeed update --pd=http://xxxxx:4199 --sink-uri="mysql://root:hxxxx@172.xxxx1:4651/" --changefeed-id="test-ticdc-tatb-task" --config ticdc_task_001.toml

6、changefeed can't resume 


[
  {
    "id": "test-ticdc-tatb-task",
    "namespace": "default",
    "summary": {
      "state": "error",
      "tso": 443433743909650434,
      "checkpoint": "2023-08-09 15:18:10.378",
      "error": {
        "time": "2023-08-09T16:21:42.632679465+08:00",
        "addr": "172.16.201.141:8335",
        "code": "CDC:ErrSnapshotTableNotFound",
        "message": "[CDC:ErrSnapshotTableNotFound]table 111 not found in schema snapshot"
      }
    }
  }
]




### What did you expect to see?

 skip the rename ddl after add Ignore-sql configuration

### What did you see instead?

[
  {
    "id": "test-ticdc-tatb-task",
    "namespace": "default",
    "summary": {
      "state": "error",
      "tso": 443433743909650434,
      "checkpoint": "2023-08-09 15:18:10.378",
      "error": {
        "time": "2023-08-09T16:21:42.632679465+08:00",
        "addr": "172.16.201.141:8335",
        "code": "CDC:ErrSnapshotTableNotFound",
        "message": "[CDC:ErrSnapshotTableNotFound]table 111 not found in schema snapshot"
      }
    }
  }
]


### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
MySQL [(none)]> SELECT tidb_version();
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                               |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v6.5.3
Edition: Community
Git Commit Hash: 71e6696f020ccf42408976d4a82cafb95717a2e4
Git Branch: heads/refs/tags/v6.5.3
UTC Build Time: 2023-06-12 13:52:32
GoVersion: go1.19.9
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

Upstream TiKV version (execute `tikv-server --version`):

```console
TiKV 
Release Version:   6.5.3
Edition:           Community
Git Commit Hash:   fd5f88a7fdda1bf70dcb0d239f60137110c54d46
Git Commit Branch: heads/refs/tags/v6.5.3
UTC Build Time:    2023-06-09 10:55:15
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

```

TiCDC version (execute `cdc version`):

```console
Release Version: v6.5.3
Git Commit Hash: 1e2f277f2e3d9b57b15db9a2a9b2c62832c071ca
Git Branch: heads/refs/tags/v6.5.3
UTC Build Time: 2023-06-05 10:19:02
Go Version: go version go1.19.9 linux/amd64
Failpoint Build: false

```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #9531 

### What is changed and how it works?
1. The `ShouldIgnoreDDLEvent()` method has been removed from the filter interface and its functionality has been integrated into the `ShouldDiscardDDL()` method. This makes the filter interface more readable and consistent.
2. DDLs can be skipped in the ddlPuller by setting `eventFilter` or `IgnoreTxnStartTs` in the filter configuration. This allows users to control which DDLs are processed, which can be useful for performance or debugging purposes. It is also helpful when we need to skip DDLs that cannot be handled by TiCDC.
3. Add a detailed error message to inform the user how to skip an DDL when it can not be handle by TiCDC.
image

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/owner/schema.go [hunk 1]
```diff
@@ -246,20 +246,43 @@ func (s *schemaWrap4Owner) BuildDDLEvents(
...
+				event.Type, event.TableInfo.TableName.Schema, event.TableInfo.TableName.Table, event.Query)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
 		}
 		if ignored {
 			s.metricIgnoreDDLEventCounter.Inc()
-			log.Info(
-				"DDL event ignored",
+			log.Panic(
```
Comment (nongfushanquan) [inline cdc/owner/schema.go:276]: warning？

Comment (asddongmen) [reply cdc/owner/schema.go]: I set it PANIC here for test purpose.  I will change it to ERROR level later.

**Pull request merged**